felix/rules/policy_test.go

```diff
@@ -507,7 +715,49 @@ var _ = Describe("Protobuf rule to iptables rule conversion", func() {
 				PreDnat:       false,
 			}
 
-			chains := renderer.PolicyToIptablesChains(policyID, policy, uint8(ipVer))
+			chains := renderer.PolicyToIptablesChains(policyID, policy, uint8(ipVer), false)
+			Expect(chains[0].Name).To(Equal("cali-pi-_d0mCmMiR44ESx5h6agZ"))
+			Expect(chains[1].Name).To(Equal("cali-po-_d0mCmMiR44ESx5h6agZ"))
+
+			inbound := chains[0].Rules
+			Expect(inbound).To(HaveLen(1))
+			Expect(inbound[0]).To(Equal(generictables.Rule{
+				Comment: []string{
+					"Policy staged:default.foo ingress",
+				},
+			}))
+
+			outbound := chains[1].Rules
+			Expect(outbound).To(HaveLen(1))
+			Expect(outbound[0]).To(Equal(generictables.Rule{
+				Comment: []string{
+					"Policy staged:default.foo egress",
+				},
```

Comment:
It seems we always add a rule with comment even when the generated rule is empty: https://github.com/projectcalico/calico/blob/bc844f6f8213d17889d1e55cd1804d0e3f7f7f2d/felix/rules/policy.go#L128